aws/sns/test_sns_pending_verified.py

```diff
@@ -1,14 +1,21 @@
 import pytest
 from helpers import get_param_id
 
-from aws.sns.resources import sns_subscription_attributes
+from aws.sns.resources import sns_subscriptions_by_topic
 
 
 @pytest.mark.sns
+@pytest.mark.rationale(
+    """
+SNS Subscriptions in PendingConfirmation status cannot receive
+notifications. They are good candidates for removal, or confirmation.
+"""
+)
 @pytest.mark.parametrize(
-    "subscription_attrs",
-    sns_subscription_attributes(),
-    ids=lambda subscription: get_param_id(subscription, "SubscriptionArn"),
+    "topic",
+    sns_subscriptions_by_topic(),
+    ids=lambda topic: get_param_id(topic, "TopicArn"),
 )
-def test_sns_pending_verified(subscription_attrs):
-    assert subscription_attrs["PendingConfirmation"] == "false"
+def test_sns_topics_without_subscriptions(topic):
+    for subscription in topic["Subscriptions"]:
+        assert subscription["SubscriptionArn"] != "PendingConfirmation"
```

Comment:
Something looks off here. Will the subscription ARN really be "PendingConfirmation"?

Reply:
Yeah, discovered that in testing.  I cannot find another way to do this.  It appears that AWS does not assign an ARN until a subscription is no longer in the PendingConfirmation state.

Reply:
My concern with this test is wondering whether it will fail frequently on newly created subscriptions? Are there cases where a subscription might go unconfirmed for a day or two? If that is the case, can we add a note about that in the rationale text?

Reply:
Or maybe we can change the test to only complain about unconfirmed subscriptions that are older than, say, a week?

Reply:
My concern with this test is wondering whether it will fail frequently on newly created subscriptions? Are there cases where a subscription might go unconfirmed for a day or two? If that is the case, can we add a note about that in the rationale text?

I don't think it will. For starters, we don't actually have any SNS subscriptions that are pending confirmation. It's conceivable that someone would create one and fail to acknowledge it in a timely fashion, but in my opinion that's an error and should be reported as such: because resources that exist, but are unusable are problematic.

As far as waiting a week, I think that delay should exist in our automation not in the test.  I also, don't think it's currently a problem we need to solve, since we don't presently have any subscriptions pending confirmation.

Reply:
I am not against adding something about it in the rationale, but I'm not sure what to put beyond what's there already.

Reply:
Yeah, I was going to recommend the "only fail after X hours", but I didn't see a created at field for subscriptions :shrug: 

>It's conceivable that someone would create one and fail to acknowledge it in a timely fashion, but in my opinion that's an error and should be reported as such: because resources that exist, but are unusable are problematic.

Ok yeah, I agree. Sounds good to me to leave as-is.